### ISSUE TITLE
Support loading from prj2epsg.org

### ISSUE THREAD

**Issue by dazza-codes** [karimbahgat/PyCRS#38]
```python
ipdb> epsg_code['url']                                                                                                                                                                                                                      
'http://prj2epsg.org/epsg/2278.json'
ipdb> pycrs.load.from_url(epsg_code['url'])                                                                                                                                                                                                 
*** pycrs.parse.FormatError: Could not auto-detect the type of crs format, make sure it is one of the supported formats
```

**Comment by karimbahgat** [comment]
The `from_url()` argument must point to a url containing a raw string of wkt or proj4. The link you're giving it is some type custom json dictionary specific to prj2epsg.org, and the wkt projection is only described inside the 'wkt' json entry. 

You can only use `from_url()` if the prj2epsg API provides a raw link to the wkt. Otherwise you would have to manually load and parse the json dictionary from the url, and then send the 'wkt' entry to pycrs. Easier would be to parse the crs code 2278 from the link itself and send it to `from_epsg()`.

**Comment by darrenleeweber** [comment]
I'm not familiar with this project and limited for time to contribute a full PR, but maybe this helps a little:

```python
def requests_retry_session(session: requests.Session = None) -> requests.Session:
    if session is None:
        session = requests.Session()
    retry = Retry(
        total=10,
        read=10,
        connect=10,
        backoff_factor=0.3,
        status_forcelist=(500, 502, 504),
    )
    https_adapter = session.get_adapter("https://")
    https_adapter.max_retries = retry
    http_adapter = session.get_adapter("http://")
    http_adapter.max_retries = retry
    return session


def wkt_to_epsg_lookup(wkt: str) -> dict:
    """Well Known Text search for matching EPSG code, using
    http://prj2epsg.org/search.json

    Arguments:
        wkt {str} -- A Well Known Text representation for a CRS

    Returns:
        [dict] -- {'code': str, 'name': str, 'url': str}
        if there is an exact match, otherwise it returns {}
    """
    session = requests_retry_session()
    search_uri = "http://prj2epsg.org/search.json"
    query = {"mode": "wkt", "terms": wkt}
    response = session.get(search_uri, params=query, timeout=60)

    if response.status_code == requests.codes.ok:
        epsg = response.json()
    else:
        logging.error("prj2epsg search error %s - %s", response.status_code, query)
        epsg = {}

    if epsg.get("exact"):
        return epsg["codes"][0]
    return {}
```

LMK if something like that could be a useful PR of some form and where that code might slot into PyCRS (there are some pytest tests for this that also use PyVCR to record the HTTP traffic).

**Comment by karimbahgat** [comment]
@darrenleeweber that would actually be a good addition. PyCRS currently can parse an EPSG code and output to some other form, but not the opposite, ie outputting to the (most likely) EPSG code, as your code above does. 

If you submit a PR I would gladly merge it! My only requests:
- Remove the dependency on 'requests' + the retry function. The same can be achieved with the builtin urllib, and we don't need the retry functionality. 
- I would probably add it as a 'wkt_to_epsg()' function in 'utils.py'. 

Later I could then probably add a 'to_epsg()' method to the toplevel CRS class that calls on this utils function.

**Comment by karimbahgat** [comment]
I guess this was fixed in #47 .
